src/xmipp/applications/tests/function_tests/test_metadata_vec_main.cpp

```diff
@@ -0,0 +1,1210 @@
+        i++;
+    }
+    ASSERT_EQ(i, mDsourceIds.size());
+}
+
+TEST_F(MetadataTest, SimilarToOperator)
+{
+    size_t id;
+    MetaDataVec auxMetadata;
+
+    ASSERT_EQ(mDsource, mDsource);
+    ASSERT_FALSE(mDsource == mDanotherSource);
+
+    // Attribute order should not be important
+    id = auxMetadata.addObject();
+    auxMetadata.setValue(MDL_Y, 2., id);
+    auxMetadata.setValue(MDL_X, 1., id);
+    id = auxMetadata.addObject();
+    auxMetadata.setValue(MDL_Y, 4., id);
+    auxMetadata.setValue(MDL_X, 3., id);
+    ASSERT_EQ(auxMetadata,mDsource);
+
+    // Test double with a given precission
+    auxMetadata.clear();
+    auxMetadata.setPrecission(2);
+    id = auxMetadata.addObject();
+    auxMetadata.setValue(MDL_Y, 2.001, id);
+    auxMetadata.setValue(MDL_X, 1., id);
+    id = auxMetadata.addObject();
+    auxMetadata.setValue(MDL_Y, 4., id);
+    auxMetadata.setValue(MDL_X, 3., id);
+    ASSERT_TRUE(auxMetadata == mDsource);
+    auxMetadata.setPrecission(4);
+    ASSERT_FALSE(auxMetadata == mDsource);
+}
+
+TEST_F(MetadataTest, AddLabel)
+{
+    MetaDataVec auxMetadata = mDunion;
+    auxMetadata.addLabel(MDL_Z);
+    std::vector<MDLabel> v1 = {MDL_X, MDL_Y, MDL_Z};
+    std::vector<MDLabel> v2 = auxMetadata.getActiveLabels();
+    EXPECT_EQ(v2, v1);
+}
+
+TEST_F(MetadataTest, AddRow)
+{
+    MetaDataVec md, md2;
+
+    MDRowVec row;
+    row.setValue(MDL_X, 1.);
+    row.setValue(MDL_Y, 2.);
+    md.addRow(row);
+    row.setValue(MDL_X, 3.);
+    row.setValue(MDL_Y, 4.);
+    md.addRow(row);
+
+    row.setValue(MDL_X, 1.);
+    row.setValue(MDL_Y, 2.);
+    md2.addRow(row);
+    row.setValue(MDL_X, 3.);
+    row.setValue(MDL_Y, 4.);
+    md2.addRow(row);
+
+    EXPECT_EQ(md, mDsource);
+    EXPECT_EQ(md2, mDsource);
+}
+
+TEST_F(MetadataTest, AddRowsPerformance)
+{
+    MetaDataVec md, md2, md3;
+    MDRowVec row;  // Sample row
+
+    printf("N_ROWS_PERFORMANCE_TEST = %d\n", N_ROWS_PERFORMANCE_TEST);
+
+    // Initialize row.
+    row.setValue(MDL_X,1.);
+    row.setValue(MDL_Y,2.);
+    row.setValue(MDL_ZSCORE,3.);
+    row.setValue(MDL_ZSCORE_HISTOGRAM,4.);
+    row.setValue(MDL_ZSCORE_RESMEAN,5.);
+    row.setValue(MDL_ZSCORE_RESVAR,6.);
+    row.setValue(MDL_ZSCORE_RESCOV,7.);
+    row.setValue(MDL_ZSCORE_SHAPE1,8.);
+    row.setValue(MDL_ZSCORE_SHAPE2,9.);
+    row.setValue(MDL_ZSCORE_SNR1,10.);
+    row.setValue(MDL_ZSCORE_SNR2,11.);
+    row.setValue(MDL_IMAGE, String("particles.stk"));
+    row.setValue(MDL_SHIFT_X_DIFF, 1.5);
+    row.setValue(MDL_SHIFT_Y_DIFF, 2.5);
+    row.setValue(MDL_CONTINUOUS_X, 1.5);
+    row.setValue(MDL_CONTINUOUS_Y, 2.5);
+    row.setValue(MDL_SHIFT_X, 1.5);
+    row.setValue(MDL_SHIFT_Y, 2.5);
+    row.setValue(MDL_SHIFT_Z, 3.5);
+
+    Timer t;
+    size_t s1, s2, s3;
+
+    t.tic();
+
+    for (int i=0; i<N_ROWS_PERFORMANCE_TEST; i++)
+    {
+        md.addRow(row);
+    }
+    s1 = t.toc("Time original:", false);
+
+    t.tic();
+    for (int i=0; i<N_ROWS_PERFORMANCE_TEST; i++)
```

Comment:
remove the second and third test, it doesn't check anything useful

Reply:
Removed.